src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,134 @@
+package frc.robot.subsystems.shooter;
+
+import com.ctre.phoenix.motorcontrol.*;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import edu.wpi.first.wpilibj.controller.LinearQuadraticRegulator;
+import edu.wpi.first.wpilibj.estimator.KalmanFilter;
+import edu.wpi.first.wpilibj.system.LinearSystem;
+import edu.wpi.first.wpilibj.system.LinearSystemLoop;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpiutil.math.Matrix;
+import edu.wpi.first.wpiutil.math.Nat;
+import edu.wpi.first.wpiutil.math.VecBuilder;
+import edu.wpi.first.wpiutil.math.Vector;
+import edu.wpi.first.wpiutil.math.numbers.N1;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.subsystems.UnitModel;
+
+import static frc.robot.Constants.Shooter.*;
+
+/**
+ * The shooter class represents the physical shooter.
+ * The purpose of the flywheel is to provide a set of functions to handle a shooting situation in a game.
+ *
+ * @author Barel
+ * @version 1.0
+ * @using TalonFX
+ * @since 2021
+ */
+public class Shooter extends SubsystemBase {
+    private final TalonFX main = new TalonFX(Ports.Shooter.MAIN);
+    private final TalonFX aux = new TalonFX(Ports.Shooter.AUX);
```

Comment:
The Engineering Team is making our lives difficult.
As you might remember, the same motors are used for the climber and the shooter. These motors already exist in the climber code: https://github.com/Galaxia5987/robot-2021/blob/c4a93a90d09358673d0c0d7e7f3a2d8c9816499f/src/main/java/frc/robot/subsystems/climber/Climber.java#L12-L13
Quoting my https://github.com/Galaxia5987/robot-2021/pull/1#discussion_r543723632:
> This gearbox and its motors are used by the shooter, too. I think it should be in a separate subsystem. You need to think how you'll make this subsystem (the gearbox) communicate effectively with both the shooter and the climber, maybe by using suppliers and lambda functions.
I'm not sure that's the best solution for this complicated situation (motors used by multiple subsystems). Consult your teammates and @orel-david.

I recommend putting the motors ("gearbox") inside another subsystem. This subsystem will include the following functionality:
1) Switching the gear
2) Setting percent output (for shooting)
3) Setting distance setpoint (for climbing)
(If you'll use state-space control in the climber too, you won't need it, but I don't think you should follow that path)
4) Getting current velocity and position from the encoder, with consideration of the current gear ratio

This subsystem will be responsible only for interaction with the hardware. The `Shooter` and `Climber` classes will change their names and will only handle "logic" and calculations.
I know it's a bit off, but I think it's the best we can do.

Reply:
@orel-david told me to concern about it later

Reply:
And I don't think we should treat it a a subsystem, I think we should treat it like a motor, with our custom functionality.

Reply:
I suggested an idea, you don't have to like it. But there is no need to be harsh.

Reply:
I meant that we create a "special" motor, that has functions to apply power to shooter and the climber(?), and the motor would automatically change the state of the "reactor" based on the motor that being used.

Reply:
Not sure what "reactor" means, but how does your idea differ from mine?

Reply:
It's not, I now realized it. The only difference is that in my version suggested creating a subclass from the motor, instead of creating a subsystem. Now I realized that it's not ideal

Reply:
I'm curious to know, what makes you think it's not "ideal"?

Reply:
To create a motor and change the state of the shifter based on the motor that is currently in use, and not through a command